services/galley/src/Galley/API/Message.hs

```diff
@@ -420,21 +420,25 @@ newUserPush p = MessagePush {userPushes = pure p, botPushes = mempty}
 newBotPush :: BotMember -> Event -> MessagePush
 newBotPush b e = MessagePush {userPushes = mempty, botPushes = pure (b, e)}
 
-runMessagePush :: ConvId -> MessagePush -> Galley ()
+runMessagePush :: Qualified ConvId -> MessagePush -> Galley ()
 runMessagePush cnv mp = do
+  localDomain <- viewFederationDomain
   pushSome (userPushes mp)
-  void . forkIO $ do
-    gone <- External.deliver (botPushes mp)
-    mapM_ (deleteBot cnv . botMemId) gone
-
-newMessageEvent :: Qualified ConvId -> Qualified UserId -> ClientId -> ByteString -> UTCTime -> ClientId -> ByteString -> Event
-newMessageEvent convId sender senderClient dat time recieverClient cipherText =
+  if localDomain /= qDomain cnv
```

Comment:
I was confused here earlier, because I didn't see that user pushes are happening regardless of whether the conversation is local. It would perhaps be helpful to add comments like:
```haskell
-- push to users
pushSome (userPushes mp)

-- push to bots (only for local conversations)
if localDomain /= qDomain cnv
...
```

Reply:
I created another little function to `pushToBots`